backend/routes/userRoutes.js

```diff
@@ -1,32 +1,29 @@
 import express from "express";
 import {
-  queryUsers,
   getUserByEmail,
   addUser,
-  deleteUser,
   getUserById,
   getUsersNonAdmins,
-  getUserRole,
   updateUser,
   getVeteransByVolunteer,
   updateUserId,
+  getVolunteersByProgram,
+  getVeteransByProgram,
 } from "../controllers/userController.js";
 import { authenticateUser } from "../middleware/auth.js";
-
+import { authenticateProfilePermissions, authenticateStaffOrAdmin, authenticateProfilePermissionsByEmail } from "../middleware/profile.js";
 const router = express.Router();
 
 // Public routes (no authentication required)
 router.post("/users", addUser); // Signup endpoint
 
 // Protected routes (authentication required)
-router.get("/users", authenticateUser, queryUsers);
-router.get("/users/email/:email", authenticateUser, getUserByEmail);
-router.get("/users/id/:id", authenticateUser, getUserById);
-router.get("/nonAdminUsers", authenticateUser, getUsersNonAdmins);
+router.get("/users/id/:userId", authenticateUser, authenticateProfilePermissions, getUserById);
+router.get("/nonAdminUsers", authenticateUser, authenticateStaffOrAdmin, getUsersNonAdmins);
 router.get("/veterans/:volunteerId", authenticateUser, getVeteransByVolunteer);
```

Comment:
Should the `GET /veterans/:volunteerId` also have an auth check to make sure the requesting user has permission to view the veterans assigned to that volunteer? (i.e. probably check they are either that volunteer themselves, or an admin)

Reply:
Thanks for adding the auth check but I'm concerned that the authenticate profile permissions middleware wouldn't work for this route since it expects a path param named userId but this route has a param named volunteerId?